File: designPattern/Composite/basic2/File.py
from Entry import Entry


class File(Entry):
    def __init__(self, name, size):
        self.size = size
        self.name = name

    def print_path(self, prefix):
        print(prefix+"/"+super().__repr__())

    def get_name(self):
        return self.name

    def get_size(self):
        return self.size

File: designPattern/visitor/review/File.java
public class File extends Entry{
	String name;
	int size;
	public File(String name, int size){
		this.name = name;
		this.size = size;
	}
	public String getName(){
		return name;
	}
	public int getSize(){
		return size;
	}
	public void accept(Visitor v){
		v.visit(this);
	}
}